tensorflow/lite/delegates/xnnpack/xnnpack_delegate.cc

```diff
@@ -210,7 +210,11 @@ class Subgraph {
     std::sort(tensors.begin(), tensors.end());
 
     // XNNPACK Value IDs for TFLite tensors
-    std::vector<uint32_t> xnnpack_tensors(tensors.back() + 1);
+    size_t tensor_size = 0;
+    if (tensors.size() > 0) {
+      tensor_size = tensors.back() + 1;
+    }
+    std::vector<uint32_t> xnnpack_tensors(tensor_size);
```

Comment:
Shouldn't `tensor_size` be `tensors.size()`? (and then potentially check that this is equal to `tensors.back() + 1`)

Imagine a scenario where `tensors` has 4 elements but the last one is 2 (e.g., it is `[0,0.1.2]` after above sorting). We make `xnnpack_tensors` have 3 elements but then the for loop below runs for 4 iterations, so it writes outside of bounds.